src/Compute/Compute/Strategies/ComputeRp/VirtualMachineScaleSetStrategy.cs

```diff
@@ -117,15 +117,23 @@ public static class VirtualMachineScaleSetStrategy
                             },
                             StorageProfile = new VirtualMachineScaleSetStorageProfile
                             {
-                                ImageReference = (imageReferenceId == null) ? imageAndOsType?.Image : (imageReferenceId.ToLower().StartsWith("/communitygalleries/") ? new ImageReference
+                                ImageReference = (imageReferenceId == null && sharedImageGalleryId == null) ? imageAndOsType?.Image 
+                                : (sharedImageGalleryId != null ? new ImageReference
                                 {
-                                    CommunityGalleryImageId = imageReferenceId,
                                     SharedGalleryImageId = sharedImageGalleryId
-                                } : new ImageReference
+                                }
+                                : (imageReferenceId.ToLower().StartsWith("/communitygalleries/") ? new ImageReference
```

Comment:
Are you sure imageReferenceId is not null? It will give a null error if fed into the ToLower() method.

Reply:
First condition checks if {{both imagereferenceId and sharedgalleryimageid} are null
{ ELSE (implies one of them is not null)
{ Check2 : sharedgalleryimage is not null 
{ ELSE (shared image gallery is null hence imagereference is not null)
}}}

Reply:
Also tested this